File: GeoBroker-Commons/src/main/java/de/hasenburg/geobroker/commons/model/message/ControlPacketType.java
package de.hasenburg.geobroker.commons.model.message;

@SuppressWarnings("SpellCheckingInspection")
public enum ControlPacketType {

	Reserved,
	CONNECT,
	CONNACK,
	PUBLISH,
	PUBACK,
	PUBREC,
	// PUBREL,
	// PUBCOMP,
	SUBSCRIBE,
	SUBACK,
	UNSUBSCRIBE,
	UNSUBACK,
	PINGREQ,
	PINGRESP,
	DISCONNECT,
	AUTH,

	// Inter-Broker Communication (no typical MQTT messages, so other spelling)
	BrokerForwardPingreq,
	BrokerForwardSubscribe,
	BrokerForwardPublish

}

File: GeoBroker-Commons/src/main/kotlin/de/hasenburg/geobroker/commons/model/message/ControlPacketType.kt
package de.hasenburg.geobroker.commons.model.message

enum class ControlPacketType {
    Reserved, //
    CONNECT, //
    CONNACK, //
    PUBLISH, //
    PUBACK, //
    PUBREC, //
    // PUBREL,
    // PUBCOMP,
    SUBSCRIBE, //
    SUBACK, //
    UNSUBSCRIBE, //
    UNSUBACK, //
    PINGREQ, //
    PINGRESP, //
    DISCONNECT, //
    AUTH, //

    // Inter-Broker Communication (no typical MQTT messages, so other spelling)
    BrokerForwardDisconnect, //
    BrokerForwardPingreq, //
    BrokerForwardSubscribe, //
    BrokerForwardUnsubscribe, //
    BrokerForwardPublish //
}
